packages/@okta/vuepress-theme-prose/components/ContentPage.vue

```diff
@@ -3,92 +3,105 @@
 </template>
 
 <script>
-import { LAYOUT_CONSTANTS } from "../layouts/Layout";
+import _ from "lodash";
+import AnchorHistory from "../mixins/AnchorHistory.vue";
 export default {
   name: "ContentPage",
+  mixins: [AnchorHistory],
   data() {
     return {
       anchors: [],
-      headingAnchorsMap: {},
-      paddedHeaderHeight: 0
+      headingAnchorsMap: {}
     };
   },
   mounted() {
-    this.paddedHeaderHeight =
-      document.querySelector(".fixed-header").clientHeight +
-      LAYOUT_CONSTANTS.HEADER_TO_CONTENT_GAP;
+    this.anchors = this.getAnchors();
     if (document.readyState === "complete") {
-        this.scrollToActiveAnchor();
-        this.captureAnchors();
+      this.onPageChange();
     } else {
       window.addEventListener("load", () => {
-        this.scrollToActiveAnchor();
-        this.captureAnchors();
+        this.onPageChange();
+      });
+      window.addEventListener("popstate", e => {
+        e.target.location.hash && this.scrollToAnchor(e.target.location.hash);
       });
     }
+
+    window.addEventListener("scroll", this.setHeadingAnchorToURL);
+  },
+  beforeDestroy() {
+    window.removeEventListener("scroll", this.setHeadingAnchorToURL);
+    window.removeEventListener("scroll", this.scrollToAnchor);
   },
   watch: {
     $page(to, from) {
-      this.$nextTick(function() {
-        this.scrollToActiveAnchor();
-        if (from.title !== to.title) {
-          this.captureAnchors();
-        }
-      });
+      if (from.title !== to.title) {
+        this.$nextTick(function() {
+          this.anchors = this.getAnchors();
+          this.getActiveAnchor(this.anchors);
+          this.onPageChange();
+        });
+      }
     }
   },
   methods: {
-    onAnchorClick(event) {
-      const element = event.target.hash
-        ? event.target
-        : event.target.closest("a");
-      if (
-        location.pathname.replace(/^\//, "") ==
-          element.pathname.replace(/^\//, "") &&
-        location.hostname == element.hostname
-      ) {
-        let scrollToAnchor = this.headingAnchorsMap[element.hash];
-        if (scrollToAnchor) {
-          event.preventDefault();
-          location.hash = element.hash;
-          return false;
-        }
+    onPageChange() {
+      const anchor = window.location.href.split("#")[1];
```

Comment:
Is there a better API to use for this? I think maybe `URL` or some other browser API can help us make this less manual.

Reply:
not sure about URL, it's not supported by all browsers. We can try to use just `window.location.hash`

Reply:
URL is polyfilled by `core-js@3` so it should be available